OptimizelySDK/Optimizely.cs

```diff
@@ -1066,6 +1067,8 @@ OptimizelyDecideOption[] options
                 }
             }
 
+            DecisionService.DecisionBatchInProgress = false;
```

Comment:
I see this will fire the saveUPS. Wondering if we have the similar fire solution for legacy APIs (activate, getVarition)

Reply:
I see this comment not pushed properly:

  We probably need one more test -
  decide(A)
  decide(B)
  2nd session picks up the UPS as expected

Reply:
> I see this comment not pushed properly:
> 
> We probably need one more test - decide(A) decide(B) 2nd session picks up the UPS as expected

Added.